### PR TITLE
Allow sorting by operating_system

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [ESSNTL-1440](https://issues.redhat.com/browse/ESSNTL-1440).
This is the change required to support the work done in this [xjoin PR](https://github.com/RedHatInsights/xjoin-search/pull/123). Allows the user to sort by the `operating_system` field.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [x] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: api/host_query_xjoin.py [hunk 1]
```diff
@@ -50,9 +50,13 @@
         }
     }
 }"""
-ORDER_BY_MAPPING = {None: "modified_on", "updated": "modified_on", "display_name": "display_name"}
-ORDER_HOW_MAPPING = {"modified_on": "DESC", "display_name": "ASC"}
-SUPPORTED_RANGE_OPERATIONS = ["gt", "gte", "lt", "lte"]
```
Comment (ryandillinfelton) [inline api/host_query_xjoin.py:55]: Why is this deleted?

Comment (kruai) [reply api/host_query_xjoin.py:55]: I deleted `SUPPORTED_RANGE_OPERATIONS` because it isn't used anywhere. Since it's a tiny change, I figured it made sense to just fix here.

Comment (RonnyPfannschmidt) [reply api/host_query_xjoin.py:55]: can we add sorting by the host id as well, it would help tests to be more stable

**Review by ryandillinfelton — DISMISSED** [review]
LGTM now that I know the deletion of `SUPPORTED_RANGE_OPERATIONS` was not a mistake :+1:

File: api/host_query_xjoin.py (hunk 1 above)
Comment (kruai) [reply api/host_query_xjoin.py:55]: @RonnyPfannschmidt I don't think this thread is the right place to request that. I'm not against adding sorting by host_id, but I also don't think a feature request should hold up this PR. If you'd like that feature, please create a Jira for it

File: api/host_query_xjoin.py [hunk 2]
```diff
@@ -50,9 +50,13 @@
...
+ORDER_BY_MAPPING = {
+    None: "modified_on",
+    "updated": "modified_on",
+    "display_name": "display_name",
+    "operating_system": "operating_system",
+}
+ORDER_HOW_MAPPING = {"modified_on": "DESC", "display_name": "ASC", "operating_system": "ASC"}
```
Comment (thearifismail) [inline api/host_query_xjoin.py:59]: Should be `"operating_system": "DESC"` so that the latest operating system is the first to appear in the list

**Review by thearifismail — COMMENTED** [review]
Commented

File: api/host_query_xjoin.py (hunk 2 above)
Comment (kruai) [reply api/host_query_xjoin.py:59]: `ORDER_HOW_MAPPING` is only for if the user tells us they want to filter on `operating_system` but doesn't specify how. But, I don't suppose it makes a big difference either way, so I can update that if you prefer

**Review by thearifismail — APPROVED** [review]

**Review by ryandillinfelton — APPROVED** [review]
LGTM

**Review by thearifismail — APPROVED** [review]
